File: BankAccountManagementSystem/src/main/java/com/project/main/service/ClientService.java
package com.project.main.service;

import java.util.ArrayList;
import java.util.NoSuchElementException;

import javax.transaction.Transactional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import com.project.main.exception.ClientNotFoundException;
import com.project.main.model.Client;
import com.project.main.repository.ClientRepository;

//This class consists of functions performed by ADMIN
@Service
@Transactional
public class ClientService {

	@Autowired
	ClientRepository clientRepository;

	// This Function is for getting the list of all Clients from the Database
	public ArrayList<Client> getAllClients() {
		ArrayList<Client> clientsList = (ArrayList<Client>) clientRepository.findAll();
		try {
			if (clientsList.isEmpty()) {
				throw new ClientNotFoundException("No Clients Found In Database");
			}
		} catch (Exception e) {
			System.out.println(e.getMessage());
		}
		return clientsList;
	}

	// This Function is for getting the details of a Client using their ClientID
	public ResponseEntity<Client> getClientById(int id) {
		try {
			Client c = clientRepository.findById(id).get();
			return new ResponseEntity<Client>(c, HttpStatus.OK);
		} catch (NoSuchElementException e) {
			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
		}
	}

	// This Function is for registering a new client
	public ResponseEntity<Client> createClient(Client client) {
		clientRepository.save(client);
		Client c = clientRepository.findById(client.getId()).get();
		return new ResponseEntity<Client>(c, HttpStatus.OK);
	}

	// This Function is for updating the details of a Client using their ClientID
	public ResponseEntity<Client> updateClient(int id, Client client) throws ClientNotFoundException {
		Client updateClient = clientRepository.findById(id)
				.orElseThrow(() -> new ClientNotFoundException("Client not exist with id: " + id));
		try {
			if (updateClient.getId() != 0) {
				updateClient.setAccountNumber(client.getAccountNumber());
				updateClient.setBalance(client.getBalance());
				updateClient.setDateOfJoining(client.getDateOfJoining());
				updateClient.setName(client.getName());
				updateClient.setOutstandingAmount(client.getOutstandingAmount());
				updateClient.setPassword(client.getPassword());
				clientRepository.save(updateClient);
				return new ResponseEntity<Client>(updateClient, HttpStatus.OK);
			} else {
				throw new ClientNotFoundException("Client Not Found In Database");
			}
		} catch (Exception e) {
			System.out.println(e.getMessage());
			return new ResponseEntity<Client>(updateClient, HttpStatus.NOT_FOUND);
		}
	}

	// This Function is for deleting a Client
	public ResponseEntity<HttpStatus> deleteClient(int id) {
		try {
			clientRepository.deleteById(id);
			return new ResponseEntity<HttpStatus>(HttpStatus.OK);
		} catch (NoSuchElementException e) {
			return new ResponseEntity<>(HttpStatus.NOT_FOUND);
		}
	}
}

File: Bank_Account_Management_System/bank_account_management_system/src/services/ClientService.js
import axios from 'axios'

const CLIENT_BASE_URL = 'http://localhost:8081/bsv'

class ClientService {

    getAllClients() {
        return axios.get(CLIENT_BASE_URL + '/clients');
    }

    createClient(client) {
        return axios.post(CLIENT_BASE_URL + '/createClient', client)
    }

    getClientById(clientId) {
        return axios.get(CLIENT_BASE_URL + '/retrieveClient/' + clientId)
    }

    updateClient(clientId, client) {
        return axios.put(CLIENT_BASE_URL + '/updateClient/' + clientId, client);
    }

    deleteClient(clientId) {
        return axios.delete(CLIENT_BASE_URL + '/deleteClient/' + clientId);
    }
    
    viewBalance(clientId, client) {
        return axios.get(CLIENT_BASE_URL + '/viewBalance/' + clientId, client);
    }

    deposit(clientId, amount) {
        return axios.put(CLIENT_BASE_URL + '/deposit/' + clientId + '/'+ amount);
    }

    withdraw(clientId, amount) {
        return axios.put(CLIENT_BASE_URL + '/withdraw/' + clientId + '/'+ amount);
    }

}

export default new ClientService();